internal/pkg/cli/validate.go

```diff
@@ -280,17 +283,51 @@ func validatePath(fs afero.Fs, val interface{}) error {
 	return nil
 }
 
-func validateStorageType(val interface{}) error {
+type validateStorageTypeOpts struct {
+	ws           manifestReader
+	workloadName string
+}
+
+func validateStorageType(val interface{}, opts validateStorageTypeOpts) error {
 	storageType, ok := val.(string)
 	if !ok {
 		return errValueNotAString
 	}
-	for _, validType := range storageTypes {
-		if storageType == validType {
-			return nil
-		}
+	if !contains(storageType, storageTypes) {
+		return fmt.Errorf(fmtErrInvalidStorageType, storageType, prettify(storageTypes))
+	}
+
+	if storageType == rdsStorageType {
+		return validateAuroraStorageType(opts.ws, opts.workloadName)
+	}
+	return nil
+}
+
+func validateAuroraStorageType(ws manifestReader, workloadName string) error {
+	if workloadName == "" {
+		return nil // Workload not yet selected while validating storage type flag.
+	}
+	mft, err := ws.ReadWorkloadManifest(workloadName)
+	if err != nil {
+		return fmt.Errorf("invalid storage type %s: read manifest file for %s: %w", rdsStorageType, workloadName, err)
+	}
+	mftType, err := mft.WorkloadType()
+	if err != nil {
+		return fmt.Errorf("invalid storage type %s: read type of workload from manifest file for %s: %w", rdsStorageType, workloadName, err)
 	}
-	return fmt.Errorf(fmtErrInvalidStorageType, storageType, prettify(storageTypes))
+	if mftType != manifest.RequestDrivenWebServiceType {
+		return nil
+	}
+	data := struct {
+		Network manifest.RequestDrivenWebServiceNetworkConfig `yaml:"network"`
+	}{}
+	if err := yaml.Unmarshal(mft, &data); err != nil {
+		return fmt.Errorf("invalid storage type %s: unmarshal manifest for %s to read network config: %w", rdsStorageType, workloadName, err)
+	}
```

Comment:
I wonder if we should move it to the `workspace` pkg like `WorkloadType()`

Reply:
I wasn't sure either, I feel like this is very niche whereas `WorkloadType` will be used more commonly that's why I've kept it here instead